redis.go

```diff
@@ -48,83 +48,57 @@ func (hs *hooks) AddHook(hook Hook) {
 func (hs hooks) process(
 	ctx context.Context, cmd Cmder, fn func(context.Context, Cmder) error,
 ) error {
-	ctx, err := hs.beforeProcess(ctx, cmd)
-	if err != nil {
-		cmd.SetErr(err)
-		return err
+	var hookIndex int
+	var retErr error
+
+	for ; hookIndex < len(hs.hooks); hookIndex++ {
+		ctx, retErr = hs.hooks[hookIndex].BeforeProcess(ctx, cmd)
+		if retErr != nil {
+			cmd.SetErr(retErr)
```

Comment:
break will not trigger idx++

Reply:
Right, but why that is a problem? I call AfterProcess for all functions except the last one that returned the error.